### PR TITLE
make default aspect DropShadows instead of Inset for UbuntuShape

### PR DESCRIPTION
**Pull request by mymike00**
this changes the default apsect of the UbuntuShape from Inset to DropShadow
**NOTE**: this will change the look of a load of components that uses the ubuntushape like buttons, switches, checkboxes, optionselectors and also the textfield in the messaging-app
I really like it but someone maybe won't. here are some screenshots

optionselector and checkboxes
optionselector and checkboxes

switches
switches

buttons
buttons

### CODE REVIEW

File: src/UbuntuToolkit/ucubuntushape.cpp
Comment (mymike00) [inline src/UbuntuToolkit/ucubuntushape.cpp:435]: is this description ok?

File: src/UbuntuToolkit/ucubuntushape.cpp
Comment (mymike00) [inline src/UbuntuToolkit/ucubuntushape.cpp:861]: not very very sure I have to change this also but I think it should...

**Comment by nfsprodriver** [comment]
I love the new optic. It bring a even more modern flare to Ubuntu Touch.

**Comment by cibersheep** [comment]
Interesting. I'm not a big fun if the Inset.
I like your changes for the buttons and textfields but not for the optionselectors and switches.

The positive aspect is that only changes the defaults but the downside is that we might want to change the behaviour on some elements.

**Comment by mymike00** [comment]
I like them also like this, but it wouldn't be so much effort changing also the components...

**Comment by cibersheep** [comment]
Ok. Let's try this in the devel. If we feel like, we can revert it or modify some components before arrives RC.
Do you think?

**Comment by mymike00** [comment]
Yeah, let's merge!

**Comment by cibersheep** [comment]
You got green light! Wweeeee... after OTA6 (which is «at the turn of the corner» meaning that will happen really soon).
So, if you don't mind waiting a little bit more, we will see your changes very soon.
THANKS!

**Comment by cibersheep** [comment]
So, has it arrive the time to try to merge this?
I see there's an error in the building tests...

**Comment by mymike00** [comment]
Yeah, how to solve them? just trigger again the build?

**Comment by UniversalSuperBox** [comment]
Thanks to https://github.com/ubports/ubuntu-ui-toolkit/pull/30 and https://github.com/ubports/ubuntu-ui-toolkit/pull/31, we finally have everything in place to build the UITK again and PRs can be merged.

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]
@NeoTheThird brought up a good point with this PR: we would need to test many apps with it before we merge. This could break the design of any app which uses the UbuntuShape, even relatively small apps like [8192](https://open-store.io/app/8192.neothethird).

An ideal test plan would be:

1. Wait for a good time in the `xenial_-_edge` cycle (now is not a good time)
2. Merge this into edge and encourage testers to try out all their apps

Alternatively, you can get people to install this PR (`sudo ubports-qa install PR-22`) and test out a lot of apps to see if any break, then report the breakage to the authors. This might be the faster solution.

Let me know what you think.

**Comment by mymike00** [comment]
I'd prefer to ask in the QA team/main supergroups for tester to try this PR, imho more people can test it and it'd faster.

**Comment by cibersheep** [comment]
Ouh. Charging my testing device :)

**Comment by cibersheep** [comment]
I have tried to install this but I can't see any change in the ui. Might it be a qrc (qml cached file?)

**Comment by mymike00** [comment]
@cibersheep Mmh that's odd. when I installed it with crossbuilder I see them after the first reboot iirc. I tried now, after an update, with `ubports-qa` but I cannot make it working...
```
$ sudo ubports-qa install ubuntu-ui-toolkit 22
CRITICAL:ubports-qa:Jenkins build not found
```

**Comment by UniversalSuperBox** [comment]
You might need to do `ubports/reponame` for some reason.

**Comment by mymike00** [comment]
I noticed that with colored ubuntu shapes you don't notice too much the drop shadows while it gave a better look than the inset shadow imho.
yeah, I agree that the option selector isn't that great but if we make it with the inset shadow it would be the only component different from others...
but, all in all, this pr doesn't introduce so drastic changes therefore I think it could be merged and then if a lot of people say to prefer some components with the old look then we can fix it later
remember that there's also another ubuntu shape aspect option, which is flat, with no shadow at all. it might be nicer for the option selector...

**Comment by dobey** [comment]
> remember that there's also another ubuntu shape aspect option, which is flat, with no shadow at all. it might be nicer for the option selector...

I think flat would probably look better for all the cases. Only the toggle switches might be a little odd, but it should already be specifying inset and outset for the shadow types of its elements, I would think.

**Comment by mymike00** [comment]
No, I don't think they do...
to decide the default aspect idk, it'd prefer dropshadow because when you use pure ubuntuShape for (usually) images/logo it fits better while other uitk components could have a specific aspect for their selves...
I'll may do some tests and see what looks nicer

**Comment by cibersheep** [comment]
I'm also a fan of flat. But I wasn't able to test this PR yet. Can you believe that?

**Comment by mymike00** [comment]
I wasn't able to install it with ubports-qa iirc, only with crossbuilder...

**Comment by mymike00** [comment]
so I tested with default aspect `Flat` and `Switch` and `OptionSelector` doesn't look nice...
we have to decide if we prefer them `Inset` or `DropShadow`, let me know
immagine
immagine
immagine

**Comment by mymike00** [comment]
what are we going to do with this?
the thing is that I don't like the Inset shadow by default (but looks good in some component like OptionSelector, Switch, CheckBox, etc...). the other 2 choices are Flat or DropShadows: usually UbuntuShape is used for images, app icons, buttons ... that have colored bg. in those situation (quite always) I think it suits better the DropShadows so I'd make it the default.
are you ok?

**Comment by cibersheep** [comment]
Sounds ok :)

**Comment by mymike00** [comment]
Now `OptionSelector`, `Switch` and `CheckBox` use bt default `Inset` aspect (which is their default look now, before this PR) while the default for everything else is DropShadow

**Comment by UniversalSuperBox** [comment]
I've reconfigured the CI so pull requests will build now. It'll be a little while until yours is finished, Jenkins is quite upset with me for that.

This PR has become quite littered with commits from `xenial`. If I can ask it of you, could you create a new PR built off of the current `xenial` branch with only your changes added? That will ensure we're only viewing and testing your changes.

**Comment by mymike00** [comment]
Sure, i'm gonna do it this week

**Pull request closed without merging**
